src/frontend/src/handler/create_source.rs

```diff
@@ -299,15 +300,28 @@ fn get_name_strategy_or_default(name_strategy: Option<AstString>) -> Result<Opti
 pub(crate) async fn bind_columns_from_source(
     session: &SessionImpl,
     source_schema: &ConnectorSchema,
-    with_properties: &BTreeMap<String, String>,
+    with_properties: Either<&WithOptions, &WithOptionsSecResolved>,
 ) -> Result<(Option<Vec<ColumnCatalog>>, StreamSourceInfo)> {
     const MESSAGE_NAME_KEY: &str = "message";
     const KEY_MESSAGE_NAME_KEY: &str = "key.message";
     const NAME_STRATEGY_KEY: &str = "schema.registry.name.strategy";
 
-    let is_kafka: bool = with_properties.is_kafka_connector();
-    let format_encode_options = WithOptions::try_from(source_schema.row_options())?.into_inner();
-    let mut format_encode_options_to_consume = format_encode_options.clone();
+    let options_with_secret = match with_properties {
+        Either::Left(options) => resolve_secret_ref_in_with_options(options.clone(), session)?,
+        Either::Right(options_with_secret) => options_with_secret.clone(),
+    };
```

Comment:
@yuhao-su  is there a reason why introducing an Either here? I see the right branch is only called in `refresh_sr_and_get_columns_diff`, why don't we need to resolve props coming from the func?

Reply:
Because that function is using a `WithOptionsSecResolved` type option from `SourceCatalog`. That means it's already been resolved before.